File: 백준/Silver/10828. 스택/스택.py
import sys

input = sys.stdin.readline

stack = []

a = int(input())

for i in range(a):
    b = input().split()
    if(b[0] == "push"):
        stack.append(b[1])
    if(b[0] == "pop"):
        print(-1 if not stack else stack.pop())
    if(b[0] == "size"):
        print(len(stack))
    if(b[0] == "empty"):
        print(0 if stack else 1)
    if(b[0] == "top"):
        print(stack[-1] if stack else -1)

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    vector<int> p;
    int a[10000];

    cin >> n;
    for(int i=0; i<n; i++) {
        string c;
        cin >> c;
        
        if(c=="push") {
            int m;
            cin >> m;
            p.push_back(m);
        } else if(c=="pop") {
            if(p.size() > 0) {
                cout << p[p.size()-1] << "\n";
                p.pop_back();
            } else {
                cout << -1 << "\n";
            }
        } else if(c=="size") {
            cout << (p.size()) << "\n";
        } else if(c=="empty") {
            cout << (p.size() == 0 ? 1 : 0) << "\n";
        } else if(c=="top") {
            cout << (p.size() ? p[p.size()-1] : -1) << "\n";
        }
    }

    return 0;
}